entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,128 @@
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToJoin)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+INNER JOIN [Posts] AS [p] ON [b].[BlogId] = [p].[BlogId]
+
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+LEFT JOIN [Posts] AS [p] ON [b].[BlogId] = [p].[BlogId]
+```
+
+### Collection selector references outer in non-predicate
+
+When collection selector has reference to outer, which isn't in predicate (as the above case), it can't be converted to join. That's why we need to evaluate collection selector for each outer element. It translates to `APPLY` operation in relational databases. If the collection is empty for outer element, then it wouldn't generate any result for that outer element. But if `DefaultIfEmpty` is applied on collection selector then outer element will be connected with default value of inner element. Because of this distinction, this kind of queries translates to `CROSS APPLY` in the absence of `DefaultIfEmpty` and `OUTER APPLY` when `DefaultIfEmpty` is applied. Sqlite doesn't support `APPLY` operator.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToApply)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+CROSS APPLY [Posts] AS [p]
+
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+OUTER APPLY [Posts] AS [p]
+```
+
+## GroupBy
+
+LINQ GroupBy operator creates result of type `IGrouping<TKey, TElement>` where `TKey` & `TElement` could be arbitrary types. Further, `IGrouping` implements `IEnumerable<TElement>`, which means you can compose over using any LINQ operator after grouping. Since no database structure can represent IGrouping, Group By operator has no translation in most cases. When an aggregate operator is applied to each group to which gives back a scalar, it can be translated to SQL `GROUP BY` in relational databases. SQL `GROUP BY` is restrictive too. It requires you to group by only scalar values  projection can contain only grouping key columns or aggregate applied over any column. EF Core identifies this subset and translates it to server. Following example shows a query, which has aggregate operator applied to it and generated SQL.
```

Comment:
```suggestion
The LINQ GroupBy operator returns results of type `IGrouping<TKey, TElement>`, where `TKey` and `TElement` could be arbitrary types. Further, `IGrouping` implements `IEnumerable<TElement>`, which means it can be composed over using any LINQ operator after grouping. Since no database structure can represent `IGrouping`, the GroupBy operator cannot be translated in most cases. As an exception, when an aggregate operator is applied to each group (e.g. MAX, COUNT), the query can be translated to  SQL `GROUP BY`. 

Note that SQL `GROUP BY` is restrictive too, requiring you to group only by scalar values; it also requires the projection to only contain grouping key columns or applied aggregate operators. EF Core identifies this subset and translates it to server, as in the following example:
```

Reply:
Acrolinx doesn't like long statements.

Reply:
Acrolinx believe that Note that does not add any value and unncessary.